lib/service/Service.ts

```diff
@@ -133,72 +76,269 @@ class Service extends EventEmitter {
+    }
   }
 
   /*
    * Subscribe to incoming peer orders.
+   * @throws ServiceError with status.INTERNAL code
    */
   public subscribePeerOrders = async (callback: Function) => {
-    this.orderBook.on('peerOrder', order => callback(order));
+    try {
+      this.orderBook.on('peerOrder', order => callback(order));
+    }catch (err) {
+      const serviceError: ServiceError = {
+        code: status.INTERNAL,
+        name: 'internal',
+        message: 'internal error occured while subscribing peer orders',
+        metadata: err,
+      };
+      throw serviceError;
+    }
   }
 
   /*
    * Subscribe to executed swaps
    */
-  public subscribeSwaps = async (_callback: Function) => {};
+  public subscribeSwaps = async (_callback: Function) => {
+  }
+
+  /*
+   * Differently from LND and RAIDEN connections this function is a must for
+   * getInfo, so it catches possible errors and converts them into meaningful ServiceError
+   * then throws it to prevent startup
+   *
+   * No logging is required since exception to be thrown will be catched and logged at upstream
+   */
+  private constructOders = async(info: any) => {
+    try {
+      const pairs = await this.orderBook.getPairs();
+      info.numPeers = this.pool.peerCount;
+      info.numPairs = pairs.length;
+
+      let peerOrdersCount: number = 0;
+      let ownOrdersCount: number = 0;
+      for (const key in pairs) {
+        const pair = pairs[key];
+
+        const [peerOrders, ownOrders] = await Promise.all([
+          this.orderBook.getPeerOrders(pair.id, 0),
+          this.orderBook.getOwnOrders(pair.id, 0),
+        ]);
+
+        peerOrdersCount += Object.keys(peerOrders.buyOrders).length + Object.keys(peerOrders.sellOrders).length;
+        ownOrdersCount += Object.keys(ownOrders.buyOrders).length + Object.keys(ownOrders.sellOrders).length;
+      }
+
+      info.orders = {
+        peer: peerOrdersCount,
+        own: ownOrdersCount,
+      };
+    } catch (err) {
+      const serviceError: ServiceError = {
+        code: status.FAILED_PRECONDITION,
+        metadata: err,
+        name: 'order constructon',
+        message: 'could not construct orders',
+      };
+      throw serviceError;
+    }
+  }
+
+  private connectToRaiden = async() => {
+    try {
+      const [address, channels] = await Promise.all([
+        this.raidenClient.getAddress(),
+        this.raidenClient.getChannels(),
+      ]);
+
+      return {
+        address,
+        channels: channels.length,
+                // Hardcoded for now until they expose it to their API
+        version: 'v0.3.0',
+      };
+
+    } catch (err) {
+      const serviceError: ServiceError = {
+        code: status.FAILED_PRECONDITION,
+        metadata: err,
+        name: 'raiden',
+        message: 'could not connect to raiden',
+      };
+      this.logger.error(`raiden error: ${serviceError}`);
+      return {
+        error: serviceError,
+      };
+    }
+  }
+
+  private connectToLnd = async() => {
```

Comment:
This does not connect to lnd, it's jsut calling getinfo, and I don't think we really need a separate method for this logic.

Reply:
@sangaman same as "Orders", getInfo method is huge, I guess we need to seperate it to handle exceptions in a better way ?